eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -1269,11 +1284,15 @@ freshKeyPair = do
   keyHash <- addKeyPair keyPair
   pure (keyHash, keyPair)
 
+-- | Generate a random KeyPair and add it to the known keys in the Imp state
 freshKeyAddr :: Era era => ImpTestM era (KeyHash r (EraCrypto era), Addr (EraCrypto era))
 freshKeyAddr = do
   keyHash <- freshKeyHash
   pure (coerceKeyRole keyHash, Addr Testnet (KeyHashObj keyHash) StakeRefNull)
 
+freshAddr :: Era era => ImpTestM era (Addr (EraCrypto era))
+freshAddr = snd <$> freshKeyAddr
```

Comment:
`freshKeyAddr` is called that way not because it returns both `KeyHash` and the `Addr`, but because it creates and `Addr` that contains a `KeyHash`, instead of a `ScriptHash`.

So this extra function is certainly helpful, it's name is far from ideal. Let's just give it a suffix of some sort instead:

```suggestion
freshKeyAddr_ :: Era era => ImpTestM era (Addr (EraCrypto era))
freshKeyAddr_ = snd <$> freshKeyAddr
```

There is a totally legitimate implementation of `freshScriptAddr` that we should implement at some point that can produce a random address for us that picks out a known spending script at random

Reply:
There are now no direct calls to `freshKeyAddr`, so I'm wondering about just changing `freshKeyAddr` to return only the `Addr`. If a caller really needs the `keyHash` they can recover it from the `Addr`. However, that requires some theoretically-failing pattern matching, so if it's likely that this could be needed in future we could keep the original function but rename it to `freshKeyAddrAndHash` or `freshKeyAddr'`. I think it's better that the more frequently used function has the shorter and more memorable name.

Reply:
I've implemented my suggestion. Let me know if you prefer what you proposed.

Reply:
That's fine. We can add one if we need it later

Reply:
Actually, spoke too soon. This becomes a breaking change and we need to bump a major version. Also I really don't like the new name: `freshKeyAddrWithHash`. Let's go with my suggestion instead